internal/kafka/internal/handlers/cluster.go

```diff
@@ -86,6 +75,23 @@ func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http
 				return nil, errors.New(errors.ErrorUnauthorized, "non admin user not authorized to perform this action")
 			}
 
+			ownershipCheckErr := provider.CheckIfOrganizationIsTheClusterOwner(orgId, clusterPayload.ClusterId, clusterSpec.ExternalID)
+			if ownershipCheckErr != nil {
+				serviceErr, ok := ownershipCheckErr.(*errors.ServiceError)
```

Comment:
`CheckIfOrganizationIsTheClusterOwner` only seems to return `ServiceError`, is this cast needed?

Reply:
`CheckIfOrganizationIsTheClusterOwner` returns the `error` interface. However, the ocm_provider implementation returns the `ServiceError` concrete implementation of that interface. Another implementation of the `CheckIfOrganizationIsTheClusterOwner` could return a different error implementation. 
 
So yes, the cast is needed since here we do not know about the `ServiceError` but the only `error` interface.